### PR TITLE
[Monitor] az monitor diagnostic-settings subscription: Support diagnositc settings for subscription

### LINKED ISSUE #13692: Subscription Diagnostic Settings for Activity Log Export
**Resource Provider**
Microsoft.Insights

**Description of Feature or Work Requested**
Support Subscription Variant of Diagnostic Settings to export Activity Log, as documented at
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/monitor/resource-manager/Microsoft.Insights/preview/2017-05-01-preview/subscriptionDiagnosticsSettings_API.json
This should look like Diagnostic Settings command, but with Subscription as target, rather than resource ID. Note, retention setting is not supported.
https://docs.microsoft.com/en-us/azure/azure-monitor/platform/diagnostic-settings
This feature will replace Add-AzLogProfile

**Minimum API Version Required**
2017-05-01-preview

**Swagger Link**
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/monitor/resource-manager/Microsoft.Insights/preview/2017-05-01-preview/subscriptionDiagnosticsSettings_API.json

**Target Date**
This was requested via ADO about 6 months ago. We need

### PR DESCRIPTION
**Pull request by MyronFanQiu**
**Description**  

Fix #13692 

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S172

File: src/azure-cli/azure/cli/command_modules/monitor/_params.py [hunk 1]
```diff
@@ -244,6 +245,16 @@ def load_arguments(self, _):
 
     with self.argument_context('monitor diagnostic-settings categories show') as c:
         c.resource_parameter('resource_uri', required=True)
+
+    with self.argument_context('monitor diagnostic-settings subscription') as c:
+        c.argument('subscription_id', validator=process_subscription_id, deprecate_info=c.deprecate(hide=True), required=False)
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/monitor/_params.py:250]: remove this this arg as it's not used.

File: src/azure-cli/azure/cli/command_modules/monitor/_params.py [hunk 2]
```diff
@@ -16,7 +16,8 @@
 from azure.cli.command_modules.monitor.util import get_operator_map, get_aggregation_map
 from azure.cli.command_modules.monitor.validators import (
     process_webhook_prop, validate_autoscale_recurrence, validate_autoscale_timegrain, get_action_group_validator,
-    get_action_group_id_validator, validate_metric_dimension, validate_storage_accounts_name_or_id)
+    get_action_group_id_validator, validate_metric_dimension, validate_storage_accounts_name_or_id,
+    process_subscription_id)
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/monitor/_params.py:20]: remove this useless import

File: src/azure-cli/azure/cli/command_modules/monitor/validators.py [hunk 3]
```diff
@@ -343,3 +343,8 @@ def validate_storage_accounts_name_or_id(cmd, namespace):
                     type='storageAccounts',
                     name=storage_account_id
                 )
+
+
+def process_subscription_id(cmd, namespace):
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/monitor/validators.py:348]: remove this useless method

**Pull request merged**